### PR TITLE
Add Architecture Roadmap

### PR DESCRIPTION
**Pull request by quinton-hoole**
Moving from Google Doc: https://docs.google.com/document/d/1VEzP20JvIKckPrGSs_R-D4vlwFzMAaqJd-9Lvfyi3jw/edit#

First commit is a straight automated conversion of the original Google Doc (minus the embedded image).  I plan to add a few more tidy-up commits before this will be ready for final review.

cc original authors @bgrant0607 @thockin @smarterclayton

### CODE REVIEW

**Comment by quinton-hoole** [comment]
Ref: https://github.com/kubernetes/community/issues/952

**Comment by quinton-hoole** [comment]
OK, I think this is ready for a final review and merge.  I'll add the opening Summary section as soon as I have a few minutes to spare.

File: contributors/devel/architectural-roadmap.md [hunk 1]
```diff
@@ -0,0 +1,1099 @@
...
+
+TBD - for those who want an overview, but don't want to read a 20 page document.
+
+## Background
+
+Kubernetes is a platform for deploying and managing containers. For
+more information about the mission, scope, and design of Kubernetes,
+see [What Is Kubernetes](http://kubernetes.io/docs/whatisk8s/) and the
+[architectural
+overview](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture.md). The
```
Comment (smarterclayton) [inline contributors/devel/architectural-roadmap.md]: Doesn't exist, is this in a PR?

File: contributors/devel/architectural-roadmap.md [hunk 2]
```diff
@@ -0,0 +1,1099 @@
...
+have. Adding to the difficulty is that, unlike some other types of
+infrastructure services such as databases, load balancers, or
+messaging systems, there are few obvious, natural boundaries for the
+“built-in” functionality. Consequently, reasonable minds can disagree
+on exactly where the boundaries lie or what principles guide the
+decisions.
+
+This document, which was inspired by [similar efforts from the
+community](https://docs.google.com/document/d/1J6yCsPtggsSx_yfqNenb3xxBK22k43c5XZkVQmS38Mk/edit),
+aims to clarify the intentions of Kubernetes’s lead architects. It is
```
Comment (smarterclayton) [inline contributors/devel/architectural-roadmap.md]: Kubernetes' architecture SIG.

**Comment by smarterclayton** [comment]
One minor comment, but I think it would be better to merge, then iterate in a set of focused steps.  Nothing in this document precludes change or the ability of the sig to alter the layering.

File: contributors/devel/architectural-roadmap.md (hunk 1 above)
Comment (quinton-hoole) [reply contributors/devel/architectural-roadmap.md]: Good catch.  It has moved to architecture/architecture.md.  I'll fix the link.

File: contributors/devel/architectural-roadmap.md (hunk 2 above)
Comment (quinton-hoole) [reply contributors/devel/architectural-roadmap.md]: Agreed, done.

**Comment by quinton-hoole** [comment]
@smarterclayton Can I ask you to push the merge button (self-merging smells bad :-) . )

**Pull request merged**